src/cookieSyncManager.ts

```diff
@@ -67,100 +70,83 @@ export default function CookieSyncManager(
 
     // Public
     this.attemptCookieSync = (
-        previousMPID: MPID,
         mpid: MPID,
         mpidIsNotInCookies?: boolean
     ): void => {
         if (!mpid || mpInstance._Store.webviewBridgeEnabled) {
             return;
         }
 
-        const { pixelConfigurations } = mpInstance._Store;
         const persistence = mpInstance._Persistence.getPersistence();
 
+        if (isEmpty(persistence)) {
+            return;
+        }
+
+        const { pixelConfigurations } = mpInstance._Store;
         pixelConfigurations.forEach((pixelSettings: IPixelConfiguration) => {
             // set requiresConsent to false to start each additional pixel configuration
             // set to true only if filteringConsenRuleValues.values.length exists
             let requiresConsent = false;
-
+            
             // Filtering rules as defined in UI
-            const { filteringConsentRuleValues } = pixelSettings;
+            const {
+                filteringConsentRuleValues,
+                pixelUrl,
+                redirectUrl,
+                moduleId,
+                // Tells you how often we should do a cookie sync (in days)
+                frequencyCap,
+            } = pixelSettings;
             const { values } = filteringConsentRuleValues || {};
-
+            
             if (!isEmpty(values)) {
                 requiresConsent = true;
             }
 
-            // Kit Module ID
-            const moduleId = pixelSettings.moduleId.toString();
-
-            // Tells you how often we should do a cookie sync (in days)
-            const frequencyCap = pixelSettings.frequencyCap;
+            // If MPID is new to cookies, we should not try to perform the cookie sync
+            // because a cookie sync can only occur once a user either consents or doesn't.
+            // we should not check if it's enabled if the user has a blank consent
+            if (requiresConsent && mpidIsNotInCookies) {
+                return;
+            }
+            
+            if (isEmpty(pixelUrl) && isEmpty(redirectUrl)) {
+                return;
+            }
 
             // Url for cookie sync pixel
-            const pixelUrl = replaceAmpWithAmpersand(pixelSettings.pixelUrl);
-
-            const redirectUrl = pixelSettings.redirectUrl
-                ? replaceAmpWithAmpersand(pixelSettings.redirectUrl)
+            const finalPixelUrl = replaceAmpWithAmpersand(pixelUrl);
+            const finalDirectUrl = redirectUrl
```

Comment:
I think these should be renamed from `final` to `modified` or `sanitized`. In my experience "final" is never "final".

Reply:
true story